### PR TITLE
fix: Added branch tag for metrics

### PR DESCRIPTION
**Pull request by JigarJoshi**
## Describe your changes
Added `branch` metrics tag. (Thanks to @pboros & @adilansari)

## How best to test these changes
Existing unit tests + manually verified on promethous endpoint `/metrics`. We will further test with debug image on dev environment.

### CODE REVIEW

File: server/config/options.go [hunk 1]
```diff
@@ -67,6 +67,8 @@ type AuthConfig struct {
 	ManagementClientId        string        `mapstructure:"management_client_id" yaml:"management_client_id" json:"management_client_id"`
 	ManagementClientSecret    string        `mapstructure:"management_client_secret" yaml:"management_client_secret" json:"management_client_secret"`
 	TokenClockSkewDurationSec int           `mapstructure:"token_clock_skew_duration_sec" yaml:"token_clock_skew_duration_sec" json:"token_clock_skew_duration_sec"`
+	GotrueUrl                 string        `mapstructure:"gotrue_url" yaml:"gotrue_url" json:"gotrue_url"`
+	GotrueUsernameSuffix      string        `mapstructure:"gotrue_username_suffix" yaml:"gotrue_username_suffix" json:"gotrue_username_suffix"`
```
Comment (pboros) [inline server/config/options.go:71]: Maybe these are from another change?

**Review by pboros — DISMISSED** [review]
One minor comment, otherwise looks good to me!

File: server/config/options.go (hunk 1 above)
Comment (JigarJoshi) [reply server/config/options.go]: ah right this is from other changes. Fixed it.

**Review by adilansari — DISMISSED** [review]

File: server/metrics/search.go [hunk 2]
```diff
@@ -33,6 +33,7 @@ func getSearchOkTagKeys() []string {
 		"env",
 		"project",
 		"db",
+		"branch",
```
Comment (ovaistariq) [inline server/metrics/search.go:36]: Should this be here?

Comment (himank) [reply server/metrics/search.go:36]: In this context, this tag is not helpful. This is used by the low-level storage, there we don't need branch-specific tags.

**Review by adilansari — APPROVED** [review]

**Pull request merged**